tests/unit/src/test/scala/org/scalafmt/tests/CoverageSuite.scala

```diff
@@ -0,0 +1,140 @@
+  checkType("A => B")               // Type.Function
+  checkType("(A, B) => C")          // Type.Function
+
+  checkType("implicit A => B")      // Type.ImplicitFunction
+  checkType("(A, B)")               // Type.Tuple
+  checkType("A with B")             // Type.With
+  checkType("A & B")                // Type.And
+  checkType("A | B", dotty)         // Type.Or
+  checkType("A { def f: B }")       // Type.Refine
+  checkType("A{}")                  // Type.Refine
+  checkType("{ def f: B }")         // Type.Refine
+
+  checkType("A forSome { type T }") // Type.Existential
+  checkType("T @A")                 // Type.Annotate
+  checkType("[X] => (X, X)")        // Type.Lambda
+  checkType("_")                    // Type.Placeholder
+  checkType("_ >: A <: B")          // Type.Bounds
+  checkType("_ <: B")               // Type.Bounds (lower)
+  checkType("_ >: A")               // Type.Bounds (uper)
+  check("def f[A <% B[A]]: C")      // Type.Bounds (view)
+  check("def f[A: B]: C")           // Type.Bounds (context)
+  check("def f[A : B : C]: D")      // Type.Bounds (context)
+
+  checkType("=> T")                 // Type.ByName
+  checkType("Any*")                 // Type.Repeated
+  check("trait A[X]")               // Type.Param
+  check("def f[@a A]: B")           // Type.Param (annotations)
+  checkPat("List[t](xs @ _*)")      // Type.Var
+
+  // meta.Term
+  check("this")                       // Term.This
+  check("a.this")                     // Term.This
+  check("a.super.b")                  // Term.Super
+  check("super[a].b")                 // Term.Super
+  check("a.super[b].c")               // Term.Super
+  // check("super")                   // Term.Super only in quasiquotes
+  check("a")                          // Term.Name
+  check("a.b")                        // Term.Select
+  check("!a")                         // Term.ApplyUnary
+  check("~a")                         // Term.ApplyUnary
+  check("-a")                         // Term.ApplyUnary
+  check("a")                          // Term.ApplyUnary
+  check("a(b)")                       // Term.Apply
+  check("a { case x => x }")          // Term.Apply
+  check("a { b }")                    // Term.Apply
+  check("a[B]")                       // Term.ApplyType
+  check("a + a")                      // Term.ApplyInfix
+  check2("a + (())")                  // Term.ApplyInfix
+  check("(a, b) + c")                 // Term.ApplyInfix
+  check("a + (b, c)")                 // Term.ApplyInfix
+  check("a + ((b, c))")               // Term.ApplyInfix
+  check("a = 1")                      // Term.Assign
+  check("return b")                   // Term.Return
+  check("throw e")                    // Term.Throw
+  check("a: A")                       // Term.Ascribe
+  check("x: @u")                      // Term.Annotate
+  check("(1, 1)")                     // Term.Tuple
+  check2("{ a; b }")                  // Term.Block
```

Comment:
what is check2? Can we give it a more meaningful name like `checkStructureOnly`?